digi/xbee/serial.py

```diff
@@ -126,8 +126,16 @@ def write_frame(self, frame):
             frame (Bytearray): The XBee API frame packet to write. If the
                 bytearray does not correctly represent an XBee frame, the
                 behaviour is undefined.
+
+        Raises:
+            CommunicationException: If there is any error writing the frame.
         """
-        self.write(frame)
+        try:
+            self.write(frame)
+        except PortNotOpenError:
+            raise CommunicationException("Serial port not open")
```

Comment:
We can keep the cause of the error with `from`:
```python
        try:
            self.write(frame)
        except PortNotOpenError as error:
            raise CommunicationException("Serial port not open") from error
        except SerialException as exc:
            raise CommunicationException(str(exc)) from exc
```

Reply:
Done.